datafusion/physical-plan/src/lib.rs

```diff
@@ -113,6 +113,15 @@ pub use datafusion_execution::{RecordBatchStream, SendableRecordBatchStream};
 /// [`required_input_distribution`]: ExecutionPlan::required_input_distribution
 /// [`required_input_ordering`]: ExecutionPlan::required_input_ordering
 pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
+    /// Short name for the ExecutionPlan, such as 'ParquetExec'.
+    fn name(&self) -> &'static str {
+        let full_name = std::any::type_name::<Self>();
```

Comment:
Just realized that users can implement their own implementations if they don't want this method to involve any computation, nice!